pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -87,9 +109,137 @@ func (pl *VolumeBinding) Filter(ctx context.Context, cs *framework.CycleState, p
 	return nil
 }
 
-// New initializes a new plugin with volume binder and returns it.
-func New(_ runtime.Object, fh framework.FrameworkHandle) (framework.Plugin, error) {
+// Reserve is called by the scheduling framework when the scheduler cache is
+// updated.
+func (pl *VolumeBinding) Reserve(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) *framework.Status {
+	allBound, err := pl.binder.AssumePodVolumes(pod, nodeName)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	cs.Write(allBoundStateKey, allBoundStateData(allBound))
+	return nil
+}
+
+// PreBind is called before binding a pod. All prebind plugins must return
+// success or the pod will be rejected and won't be sent for binding.
+//
+// PreBind will make the API update with the assumed bindings and wait until
+// the PV controller has completely finished the binding operation.
+//
+// If binding errors, times out or gets undone, then an error will be returned to
+// retry scheduling.
+func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) *framework.Status {
+	state, err := cs.Read(allBoundStateKey)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	allBound, ok := state.(allBoundStateData)
+	if !ok {
+		return framework.NewStatus(framework.Error, "unable to convert state into allBoundStateData")
+	}
+	if allBound {
+		// no need to bind volumes
+		return nil
+	}
+	klog.V(5).Infof("Trying to bind volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	err = pl.binder.BindPodVolumes(pod)
+	if err != nil {
+		klog.V(1).Infof("Failed to bind volumes for pod \"%v/%v\": %v", pod.Namespace, pod.Name, err)
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	klog.V(5).Infof("Success binding volumes for pod \"%v/%v\"", pod.Namespace, pod.Name)
+	return nil
+}
+
+// Unreserve is called by the scheduling framework when a reserved pod was
+// rejected in a later phase.
+// TODO(cofyc) Revert assumed PV/PVC cache, see http://issues.k8s.io/82934#issuecomment-538269188
+func (pl *VolumeBinding) Unreserve(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) {
+	pl.binder.DeletePodBindings(pod)
+	return
+}
+
+// PostBind is called after a pod is successfully bound.
+func (pl *VolumeBinding) PostBind(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) {
+	pl.binder.DeletePodBindings(pod)
+	return
+}
+
+// Args holds the args that are used to configure the plugin.
+type Args struct {
+	BindTimeoutSeconds *int64 `json:"bindTimeoutSeconds,omitempty"`
+}
```

Comment:
And its versioned counterparts.

Reply:
hmm, I didn't mentioned the versioned counterparts because I wanted it to be done in the same issue for deprecating bindTimeoutSeconds in CC. The goal was to separate the API issue from the implementation issue (the focus of this PR). WDYT?